File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    var answer = 0;
    people.sort((a,b) => b-a)
    let st = 0
    let en = people.length - 1
    while (st <= en){
        if (st == en){
            answer ++
            break
        }
        if(people[st] + people[en] <= limit){
            en --
        }
        st ++
        answer ++
    }
    return answer;
}

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        Integer[] arrPeople = new Integer[people.length];
        for (int i=0; i<arrPeople.length; i++) arrPeople[i] = people[i];
        Arrays.sort(arrPeople, Collections.reverseOrder());
        int l = 0;
        int r = arrPeople.length - 1;
        
        while (l < r) {
            int left = arrPeople[l];
            int right = arrPeople[r];
            if (left + right <= limit) {
                answer++;
                l++;
                r--;
                continue;
            }
            answer++;
            l++;
        }
        if (l == r) {
            answer++;
        }
        
        return answer;
    }
}